### PR TITLE
#373 competencies service should triage competencies based on categories

### PR DESCRIPTION
**Pull request by Nazanin-Rezaei**
## Proposed changes
#373
create two functions to getAllcompetenciesByCategory  and listCategories.
To test the functionality of the functions, I console.log the result of functions and I got the inputs from my database.
Also, I create two tests inside of competencyServices test file and checked them with the yarn test in the terminal.  

## Checklist

- [x] Are the issues being addressed linked to this PR?
- [x] Do all commit messages to start with the issue number?
- [x] Are all code changes sufficiently tested?

### CODE REVIEW

File: api/src/services/competenciesService.ts [hunk 1]
```diff
@@ -55,3 +55,20 @@ export const authorizeCompetencyUpdate = async (
 
   return !!competency;
 };
+
+export const getAllcompetenciesByCategory = async (categoryId: number) => {
```
Comment (MahoMori) [inline api/src/services/competenciesService.ts:59]: I think "competencies" in the function name should be capitalized.
Current: **getAllcompetenciesByCategory**
Change to: **getAllCompetenciesByCategory**

File: api/src/services/__tests__/competenciesService.ts [hunk 2]
```diff
@@ -107,4 +109,48 @@ describe('competenciesService', () => {
       ).toEqual(false);
     });
   });
+
+  describe('getAllcompetenciesByCategory', () => {
+    it('should return competencies base on the categori_id', async () => {
```
Comment (achichikalova) [inline api/src/services/__tests__/competenciesService.ts:114]: typo: category_id

**Review by achichikalova — CHANGES_REQUESTED** [review]
It needs some changes to be done before approval.

File: api/src/services/competenciesService.ts [hunk 3]
```diff
@@ -55,3 +55,20 @@ export const authorizeCompetencyUpdate = async (
...
+  return competencies;
+};
+
+export const listCategories = async (limit: number, offset: number) => {
+  const categories = await db('categories')
+    .select('id', 'label', 'description', 'image_url')
+    .orderBy('label', 'asc')
+    .orderBy('id', 'asc')
+    .limit(limit)
+    .offset(offset);
```
[the remark this answers is not in the document]
Comment (Nazanin-Rezaei) [reply api/src/services/competenciesService.ts:72]: I was not able to get the testing approval without the limit and offset features. That is why I add them back to the function!

[the remark this answers is not in the document]
Comment (sheacronin) [reply api/src/services/competenciesService.ts:72]: Did you already try changing the test to not have the limit or offset bindings in the mockQuery? I just don't want us to have to add the limit or offset if we won't be using them.

**Review by sheacronin — APPROVED** [review]
It appears that both of these functions do what we need them to do, getting all the categories and also getting all competencies with a category id. The test functions appear to correctly test the two added functions.

**Comment by Ayako-Yokoe** [comment]
Great work, Naz.
I just merged #371, so it might need updating.

**Pull request merged**